### ISSUE TITLE
no-relative-import-paths plugin not reported as error by the extension in some cases

### ISSUE THREAD

**Issue by jgoux** [microsoft/vscode-eslint#1594]
Hello,

I'm using https://github.com/MelvinVermeer/eslint-plugin-no-relative-import-paths and I'm having trouble having vscode-eslint reporting errors in my import statements.

Running eslint CLI against my file works as expected, the error is seen and fixable, but somehow vscode-eslint is missing it.

What's even more strange is that if I change the plugin's options, vscode-eslint is able to pick up the rule. But the combination that I want to use isn't detected.

With the CLI:
<img width="979" alt="image" src="https://user-images.githubusercontent.com/1443499/215732468-818391c0-cc71-4dd5-a0f1-c2038d1c93db.png">

You can see that vscode-eslint isn't seeing the issue (line 3 should be in error).

If I change the options of the plugin from:
```js
{
"no-relative-import-paths/no-relative-import-paths": [
      "error",
      { "allowSameFolder": true, "rootDir": "src", "prefix": "~" },
    ]
}
```

to:

```js
{
"no-relative-import-paths/no-relative-import-paths": [
      "error",
      { "allowSameFolder": true, "rootDir": "", "prefix": "~" },
    ]
}
```

The vscode extension is now able to see the error:

<img width="923" alt="image" src="https://user-images.githubusercontent.com/1443499/215733338-21fcc3a3-d1c2-4ac8-a687-dc3605195299.png">

It seems that no matter what I put in `rootDir`, if it's not an empty string, vscode-eslint won't report the error (but the CLI works as expected, and there is no open issue about the plugin not working as I intend to use it in their repository).

It's the only rule acting this way among my numerous other rules. 😅

The extension is using my local eslint library installed in my monorepo:
`[Info  - 11:28:27] ESLint library loaded from: /Users/jgoux/Documents/code/snaplet-labs/node_modules/eslint/lib/api.js`

**Comment by dbaeumer** [comment]
@jgoux this is very likely a problem of a correct working directory setup. Looks like validation happens correctly in the terminal when in the api directory. Such a working directory must be know to the extension as well. See the setting `eslint.workingDirectories`

**Comment by dbaeumer** [comment]
Looks ok. Can you please provide me with a Github repository I can clone that demos what you are seeing. This ensures that we both look at the same setup.

**Comment by jgoux** [comment]
@dbaeumer Sorry for the delay! Here is a reproduction: https://github.com/jgoux/vscode-eslint-imports

**Comment by dbaeumer** [comment]
The underlying problem is that the plugin does path math without using the path npm module which results in a path with `//` which then doesn't correctly match. See https://github.com/MelvinVermeer/eslint-plugin-no-relative-import-paths/issues/23

**Comment by dbaeumer** [comment]
See also https://github.com/microsoft/vscode-eslint/pull/1605 to avoid these situations as well.

**Comment by jgoux** [comment]
Thanks for tracking it down!

Is your PR fixing the issue or do we have to fix the issue in https://github.com/MelvinVermeer/eslint-plugin-no-relative-import-paths/issues/23 ?

**Comment by dbaeumer** [comment]
Mine works around it but https://github.com/MelvinVermeer/eslint-plugin-no-relative-import-paths/issues/23 should be addressed as well.
